core/src/iam/signin.rs

```diff
@@ -156,13 +157,41 @@ pub async fn db_access(
 												id: Some(rid.to_raw()),
 												..Claims::default()
 											};
+											// AUTHENTICATE clause
+											if let Some(ac) = at.authenticate {
+												// Setup the system session for finding the signin record
+												let mut sess =
+													Session::editor().with_ns(&ns).with_db(&db);
+												sess.rd = Some(rid.clone().into());
+												sess.tk = Some(claims.clone().into());
+												sess.ip.clone_from(&session.ip);
+												sess.or.clone_from(&session.or);
+												// Compute the value with the params
+												match kvs.evaluate(ac, &sess, None).await {
+													Ok(val) => match val.record() {
+														Some(id) => {
+															// Update rid with result from AUTHENTICATE clause
+															rid = id;
+															claims.id = Some(rid.to_raw());
```

Comment:
See previous comment.

Reply:
I addressed this in [a54b5bd](https://github.com/surrealdb/surrealdb/pull/4161/commits/a54b5bd5c4a8b4817a71b8f90a2cf8ba87ea0949). Let me know what you think!